hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -556,7 +570,9 @@ private static void processRollbackMetadata(HoodieActiveTimeline metadataTableTi
       //
       // when at time t4, we commit the compaction rollback,the above check returns true.
       HoodieInstant syncedInstant = new HoodieInstant(false, HoodieTimeline.DELTA_COMMIT_ACTION, instantToRollback);
-      if (metadataTableTimeline.getCommitsTimeline().isBeforeTimelineStarts(syncedInstant.getTimestamp())) {
+      Option<HoodieInstant> firstInstant = metadataTableTimeline.getCommitsTimeline().firstInstant();
```

Comment:
there could be more cases to consider here :( 
if archival of MDT kicks in, then the firstInstant will refer to some other future commit. so, the fix may not work in all cases.

Reply:
Right.  I actually thought about that, but that requires more changes and touches the archived timeline as well.  Let's keep the scope of this PR and fix that archival-related case later?